python/taichi/lang/__init__.py

```diff
@@ -77,6 +77,14 @@ def init(default_fp=None, default_ip=None, print_preprocessed=None, debug=None,
 
   for k, v in kwargs.items():
     setattr(ti.cfg, k, v)
+
+  if os.environ.get("TI_PRINT_IR") is not None:
+    ti.print_ir = True
+  arch = os.environ.get("TI_ARCH")
+  if arch is not None:
+    ti.cfg.arch = getattr(core.Arch, arch)
```

Comment:
`arch` is a string and should be translated to e.g. `ti.cuda`.

Reply:
You mean, use a `dict()`?

Reply:
Added `arch_from_name`: 96b116ee